### DISCUSSION TITLE
[@types/react] FunctionalComponent type is not returning PropsWithChildren at props

### DISCUSSION THREAD

**Discussion by gmass0n** [DefinitelyTyped/DefinitelyTyped#60114] [Issues with a @types package]
After updrage the version to 18, the type FunctionalComponent it's without children at props

[a comment by an automated account is not shown]

**Comment by MysteryBlokHed** [comment]
I think this was done deliberately with the types for React 18. Manually adding a `children` key to your component's props with the type `React.ReactNode` or `React.ReactNode[]` should fix the problem.

**Reply by hellatan** [reply]
yeeep...not a fun upgrade but it's better to be more deliberate than having a type mask your props.

pre-react 18:
```js
// FC is the same as FunctionComponent
const Component: FC = ({ children }) => {
    return <div>{children}</div>;
};
```
react 18:
```js
const Component: FC<{ children: ReactNode }> = ({ children }) => {
    return <div>{children}</div>;
};
```
I don't think you need `ReactNode[]` b/c `ReactNode` has `ReactFragment` as apart of its type definition which is 
```ts
type ReactFragment = Iterable<ReactNode>;
```
There is also the `PropsWithChildren` type which you can use like the following:
```ts
const Component: PropsWithChildren<FC> = ({ children }) => {
    return <div>{children}</div>;
}
```
but `PropsWithChildren` makes `children` optional: 
```ts
type PropsWithChildren<P> = P & { children?: ReactNode | undefined };
```

**Comment by eps1lon** [comment]
This change was intended. Check out https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210 for a full list of intended, breaking changes and their rationale. The PR includes links to a codemod that helps migrating to `@types/react@18`
